junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/descriptor/JupiterTestDescriptor.java

```diff
@@ -81,7 +81,7 @@
 
 	@Override
 	public boolean isLeaf() {
-		return !isContainer();
+		return isTest();
```

Comment:
That's why they, namely `TestFactoryTestDescriptor` and `TestTemplateTestDescriptor`, do override that method with `return true;`